views/widget.html

```diff
@@ -6,12 +6,30 @@
 <label for="{{ id }}" class="apos-forms-label {{ classPrefix + '__label' if classPrefix }}">
   {{ widget.fieldLabel}}
   {% if not widget.required %} {{ __('(Optional)') }}{% endif %}
-  <span class="apos-forms-label-message {{ classPrefix + '__label-message' if classPrefix }}" data-apos-input-message="{{ widget.fieldName}}" hidden></span>
+  <span class="apos-forms-label-message {{ classPrefix + '__label-message' if classPrefix }}" data-apos-input-message="{{ widget.fieldName }}" hidden></span>
 </label>
 
-<input class="apos-forms-input {{ classPrefix + '__input' if classPrefix }}"
+<input class="apos-forms-input {{ classPrefix + '__input' if classPrefix }} apos-forms-input-{{ addressPart }}"
   type="string"
-  id="{{ id }}" name="{{ widget.fieldName}}"
-  placeholder="{{ widget.placeholder}}"
-  {% if widget.required %}required{% endif %}
+  id="{{ id }}" name="{{ widget.fieldName }}"
+  placeholder="{{ widget.placeholder }}"
+  data-apos-forms-google-api-key="{{ data.googleApiKey }}"
+  {% if widget.required -%} required {%- endif -%}
 />
+
+{% if widget.splitAddress and widget.addressParts %}
+  {% for addressPart in widget.addressParts %}
+    {%- if widget.displaySplitAddress %} </br> {%- endif %}
+    <label for="{{ addressPart }}" class="apos-forms-label {{ classPrefix + '__label' if classPrefix }}" {%- if not widget.displaySplitAddress %} style="display: none;" {%- endif %}>
+      {{ __(addressPart) }}
+      <span class="apos-forms-label-message {{ classPrefix + '__label-message' if classPrefix }}" data-apos-input-message="{{ addressPart }}" hidden></span>
+    </label>
+    <input class="apos-forms-input {{ classPrefix + '__input' if classPrefix }} apos-forms-input-{{ addressPart }}"
```

Comment:
Should these fields be disabled? Someone could edit the split fields after they have been populated. Then the form data could have conflicting addresses. Please let me know if I'm missing something.

Reply:
Google is pretty darn smart, but still the user knows where they live better than Google does. If it miscategorizes some of their input it makes sense that it is possible to correct it.

Reply:
That's reasonable, but are we still sending back two versions of the address?

Reply:
Sure, but different users of the spreadsheet might appreciate copying and pasting an address vs. sorting by zip code at different times.

Reply:
That doesn't change the fact that you can easily have two different addresses and not know which is the correct one.

Reply:
I don't think I understand your point very well.

Reply:
We're potentially sending two sets of conflicting data: the full address AND split address fields that may have been altered after this module populated them. Maybe both forms are useful, but both conflicting versions?

Reply:
In fact, no. I have just realized it is not possible to modify the address even if the inputs are not disabled.

I have typed "107, rue Servient, Lyon" and modified the city in the full input, and the street number in the broken out fields.
![image](https://user-images.githubusercontent.com/11479686/89516468-c2d70280-d7d8-11ea-83cc-2906d4862dc1.png)

But here is what is saved in DB:
![image](https://user-images.githubusercontent.com/11479686/89516599-ee59ed00-d7d8-11ea-87d7-7bb2747eab2f.png)

@boutell @abea Is it a problem? Should we respect user's input after Google search result?

Reply:
I know why. The broken out fields values are updated after each Google search (so when the full address input triggers a new search). Values are then taken from the search and not from the inputs. Should it be changed?

Reply:
Good catch. The UX issue remains. I suggest that if someone edits one of the split fields, then the unified address field will be reset (emptied) and the value not submitted. If the person enters a new address in that unified address field, then the split fields are repopulated (emptied first). Repeat as necessary.